sys/kern/sysv_shm.c

```diff
@@ -389,24 +401,25 @@ struct shmdt_args {
 int
 sys_shmdt(struct thread *td, struct shmdt_args *uap)
 {
-
-	return (kern_shmdt(td, uap->shmaddr));
-}
-
-static int
-kern_shmdt(struct thread *td, const void * __capability shmaddr)
-{
-	int error;
+	const void * __capability shmaddr = uap->shmaddr;
 
 #if __has_feature(capabilities)
 	/*
-	 * Check for VMMAP unconditionally as non-CheriABI processes
-	 * will have added it in __USER_CAP.
+	 * Require a valid, unsealed, VMMAP bearing capability or NULL.
+	 * length is checked after we find our mapping.
 	 */
 	if (shmaddr != NULL &&
-	    (cheri_getperm(shmaddr) & CHERI_PERM_CHERIABI_VMMAP) == 0)
+	    (!cheri_gettag(shmaddr) || cheri_getsealed(shmaddr) ||
```

Comment:
We should add the sealed check in `sys_mmap` and `sys_munmap` as well.

Reply:
I think so and not yet.